lib/npo.src.js

```diff
@@ -12,12 +12,59 @@
 	/*jshint validthis:true */
 	"use strict";
 
-	var builtInProp, cycle, scheduling_queue,
+	var builtInProp, cycle, scheduling_queue, nextShouldNotEmit = false,
 		ToString = Object.prototype.toString,
 		timer = (typeof setImmediate != "undefined") ?
 			function timer(fn) { return setImmediate(fn); } :
-			setTimeout
+			setTimeout,
+        //Adapted from [when](https://github.com/cujojs/when/blob/master/lib/makePromise.js)
+        emit = (function () {
+			function noop(){}
+			if (typeof process === 'object' && typeof process.emit === 'function') {
```

Comment:
code style in this file uses `"`, not `'`, for string literals. let's stay consistent.